x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/security/authz_api_keys/10_index_authz.yml

```diff
@@ -0,0 +1,343 @@
+---
+setup:
+  - skip:
+      features: headers
+
+  - do:
+      cluster.health:
+          wait_for_status: yellow
+
+  - do:
+      indices.create:
+        index: only_read
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+
+  - do:
+      indices.create:
+        index: only_create_doc
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+
+  - do:
+      indices.create:
+        index: only_index
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+
+  - do:
+      indices.create:
+        index: only_delete
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+
+  - do:
+      indices.create:
+        index: everything
+        body:
+          mappings:
+            properties:
+              name:
+                type: "keyword"
+
+---
+"Test indexing documents with API Key":
+  - do:
+      security.create_api_key:
+        body:
+          name: "test-index-access"
+          expiration: "1d"
+          role_descriptors:
+            index_access:
+              indices:
+                - names: [ "only_read" ]
+                  privileges: [ "read" ]
+                - names: [ "only_create_doc" ]
+                  privileges: [ "create_doc" ]
+                - names: [ "only_index" ]
+                  privileges: [ "index" ]
+                - names: [ "only_delete" ]
+                  privileges: [ "delete" ]
+                - names: [ "everything" ]
+                  privileges: [ "all" ]
+  - match: { name: "test-index-access" }
+  - is_true: id
+  - set:
+      id: api_key_id
+      encoded: credentials
+
+  ###
+  # Successful indexing
+  ###
+
+  # document create
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      create:
+        id: "c1"
+        index: only_create_doc
+        body: >
+          {
+            "name" : "create-doc-1"
+          }
+
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      create:
+        id: "i1"
+        index: only_index
+        body: >
+          {
+            "name" : "index-doc-1"
+          }
+
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      create:
+        id: "e1"
+        index: everything
+        body: >
+          {
+            "name" : "everything-doc-1"
+          }
+
+  # document index
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      index:
+        # no "id" due to create_doc privilege
```

Comment:
You can specify `id` along with `op_type=create`.

Reply:
Yes you can, but then it's really just testing `create` again.